inference/core/workflows/core_steps/analytics/path_deviation/v1.py

```diff
@@ -0,0 +1,176 @@
+    detections: StepOutputSelector(
+        kind=[
+            OBJECT_DETECTION_PREDICTION_KIND,
+            INSTANCE_SEGMENTATION_PREDICTION_KIND,
+        ]
+    ) = Field(  # type: ignore
+        description="Predictions",
+        examples=["$steps.object_detection_model.predictions"],
+    )
+    triggering_anchor: Union[str, WorkflowParameterSelector(kind=[STRING_KIND]), Literal[tuple(sv.Position.list())]] = Field(  # type: ignore
+        description=f"Triggering anchor. Allowed values: {', '.join(sv.Position.list())}",
+        default="CENTER",
+        examples=["CENTER"],
+    )
+    reference_path: Union[list, StepOutputSelector(kind=[LIST_OF_VALUES_KIND]), WorkflowParameterSelector(kind=[LIST_OF_VALUES_KIND])] = Field(  # type: ignore
+        description="Reference path in a format [(x1, y1), (x2, y2), (x3, y3), ...]",
+        examples=["$inputs.expected_path"],
+    )
+
+    @classmethod
+    def describe_outputs(cls) -> List[OutputDefinition]:
+        return [
+            OutputDefinition(
+                name=OUTPUT_KEY,
+                kind=[
+                    OBJECT_DETECTION_PREDICTION_KIND,
+                    INSTANCE_SEGMENTATION_PREDICTION_KIND,
+                ],
+            ),
+        ]
+
+    @classmethod
+    def get_execution_engine_compatibility(cls) -> Optional[str]:
+        return ">=1.0.0,<2.0.0"
+
+
+class PathDeviationAnalyticsBlockV1(WorkflowBlock):
+    def __init__(self):
+        self._object_paths: Dict[
+            str, Dict[Union[int, str], List[Tuple[float, float]]]
+        ] = {}
+
+    @classmethod
+    def get_manifest(cls) -> Type[WorkflowBlockManifest]:
+        return PathDeviationManifest
+
+    def run(
+        self,
+        detections: sv.Detections,
+        metadata: VideoMetadata,
+        triggering_anchor: str,
+        reference_path: List[Tuple[int, int]],
+    ) -> BlockResult:
+        if detections.tracker_id is None:
+            raise ValueError(
+                f"tracker_id not initialized, {self.__class__.__name__} requires detections to be tracked"
+            )
+
+        video_id = metadata.video_identifier
+        if video_id not in self._object_paths:
+            self._object_paths[video_id] = {}
+
+        anchor_points = detections.get_anchors_coordinates(anchor=triggering_anchor)
+        result_detections = []
+        for i, tracker_id in enumerate(detections.tracker_id):
+            detection = detections[i]
+            anchor_point = anchor_points[i]
+            if tracker_id not in self._object_paths[video_id]:
+                self._object_paths[video_id][tracker_id] = []
+            self._object_paths[video_id][tracker_id].append(anchor_point)
+
+            object_path = np.array(self._object_paths[video_id][tracker_id])
+            ref_path = np.array(reference_path)
+
+            frechet_distance = self._calculate_frechet_distance(object_path, ref_path)
+            detection[DETECTIONS_PATH_DEVIATION_PARAM] = np.array(
```

Comment:
@grzegorz-roboflow please ack this

that's change in the direction of extending our "extended metadata" interface for `sv.Detections`. I believe this is ok, yet for this field to be serialised we need to add change [here](https://github.com/roboflow/inference/blob/e7578d5f4729462a16a688df1fa35c885f0bf537/inference/core/workflows/core_steps/common/serializers.py#L29)